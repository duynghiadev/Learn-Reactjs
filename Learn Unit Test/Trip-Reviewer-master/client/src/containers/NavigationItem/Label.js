import styled from 'styled-components';
import PropTypes from 'prop-types';

const getCalculatedStyles = ({ sidenav, limit }) => `
  @media (max-width: ${limit}px) {
    margin-left: ${sidenav ? '10px' : '0'}
  }
`;

const Label = styled.span`
  margin-left: 10px;

  &:hover {
    color: white;
  }

  ${props => getCalculatedStyles(props)}
`;

Label.propTypes = {
  limit: PropTypes.number,
  sidenav: PropTypes.number,
};

Label.defaultProps = {
  limit: 0,
  sidenav: 0,
};

export default Label;
