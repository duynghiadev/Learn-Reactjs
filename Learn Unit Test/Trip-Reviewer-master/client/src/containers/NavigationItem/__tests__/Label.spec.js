import React from 'react';
import { create } from 'react-test-renderer';

import Label from '../Label';

const mockComponent = props => <Label {...props} />;

describe('NavigationItem/<Label />', () => {
  it('should render regular component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render sidenav component', () => {
    const props = { sidenav: true };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
