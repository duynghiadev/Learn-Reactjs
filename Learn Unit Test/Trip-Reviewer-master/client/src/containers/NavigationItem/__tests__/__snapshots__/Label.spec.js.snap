// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem/<Label /> should render regular component 1`] = `
<span
  className="Label-bxMjsr dewtPP"
/>
`;

exports[`NavigationItem/<Label /> should render sidenav component 1`] = `
<span
  className="Label-bxMjsr ekgpif"
/>
`;
